### ISSUE TITLE
Conv2DCustomBackpropFilterOp only supports NHWC

### ISSUE THREAD

**Issue by ljppro** [NervanaSystems/ngraph-tf#91]
Hi,

When I run tf_cnn_benchmarks.py, it reports that error "Conv2DCustomBackpropFilterOp only supports NHWC" as following:

2018-07-20 00:56:31.727280: E tensorflow/core/common_runtime/executor.cc:696] Executor failed to create kernel. Invalid argument: Conv2DCustomBackpropFilterOp only supports NHWC.
         [[Node: v0/tower_0/gradients/v0/tower_0/cg/resnet_v115/conv52/conv2d/Conv2D_grad/Conv2DBackpropFilter = Conv2DBackpropFilter[T=DT_FLOAT, data_format="NCHW", dilations=[1, 1, 1, 1], padding="SAME", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](ngraph_cluster_0/_1691, ConstantFolding/v0/tower_0/gradients/v0/tower_0/cg/resnet_v115/conv52/conv2d/Conv2D_grad/ShapeN-matshapes-1, v0/tower_0/gradients/v0/tower_0/cg/resnet_v115/conv52/batchnorm52/FusedBatchNorm_grad/FusedBatchNormGrad)]]

Does it mean ngraph-tf only supports CPU?

**Comment by aprocter** [comment]
Hi there,

The short answer is that at the moment training will probably not work with this script through `ngraph-tf`, because gradient/backprop ops are generally mostly unimplemented. Support for training should be coming soon. It is also true that at the moment, `ngraph-tf` only supports nGraph's CPU backend, but again, support for more backends will be coming soon.

In a bit more detail: the directions recommend that you use the NCHW format, because that is what works best on nGraph. However, since `ngraph-tf` doesn't yet support the `Conv2DBackpropFilter` op, placement for that op falls back on the standard CPU implementation, and the standard CPU implementation only supports NHWC. As such, training _might_ be functional if you switch to NHWC, but unfortunately it will not be performant.
